src/hexkit/providers/akafka/provider/eventsub.py

```diff
@@ -266,3 +455,233 @@ async def run(self, forever: bool = True) -> None:
+
+    Args:
+    - `event`: The event to process.
+
+    Returns:
+    - `ConsumerEvent`: The unaltered event to publish to the retry topic.
+    - `None`: A signal to discard the event.
+
+    Raises:
+    - `DLQValidationError`: If the event headers are invalid.
+    """
+    validate_dlq_headers(event)
+    return ExtractedEventInfo(event)
+
+
+class KafkaDLQSubscriber(InboundProviderBase):
+    """A kafka event subscriber that subscribes to the configured DLQ topic and either
+    discards each event or publishes it to the retry topic as instructed.
+    Further processing before requeuing is provided by a callable adhering to the
+    DLQEventProcessor definition.
+
+    """
+
+    @classmethod
+    @asynccontextmanager
+    async def construct(
+        cls,
+        *,
+        config: KafkaConfig,
+        dlq_publisher: EventPublisherProtocol,
+        process_dlq_event: DLQEventProcessor = process_dlq_event,
+        kafka_consumer_cls: type[KafkaConsumerCompatible] = AIOKafkaConsumer,
+    ):
+        """
+        Setup and teardown KafkaDLQSubscriber instance.
+
+        Args:
+        - `config`:
+            Config parameters needed for connecting to Apache Kafka.
+        - `dlq_publisher`:
+            running instance of publishing provider that implements the
+            EventPublisherProtocol, such as KafkaEventPublisher. It is used to publish
+            events to the configured retry topic.
+        - `kafka_consumer_cls`:
+            Overwrite the used Kafka consumer class . Only intended for unit testing.
+        - `process_dlq_event`:
+            An async callable adhering to the DLQEventProcessor definition that provides
+            validation and processing for events from the DLQ. It should return _either_
+            the event to publish to the retry topic (which may be altered) or `None` to
+            discard the event. The `KafkaDLQSubscriber` will log and interpret
+            `DLQValidationError` as a signal to discard/ignore the event, and all other
+            errors will be re-raised as a `DLQProcessingError`.
+        """
+        client_id = generate_client_id(
+            service_name=config.service_name, instance_id=config.service_instance_id
+        )
+
+        consumer = kafka_consumer_cls(
+            config.kafka_dlq_topic,
+            bootstrap_servers=",".join(config.kafka_servers),
+            security_protocol=config.kafka_security_protocol,
+            ssl_context=generate_ssl_context(config),
+            client_id=client_id,
+            group_id=config.service_name,
+            auto_offset_reset="earliest",
+            enable_auto_commit=False,
+            key_deserializer=lambda event_key: event_key.decode("ascii"),
+            value_deserializer=lambda event_value: json.loads(
+                event_value.decode("ascii")
+            ),
+            max_partition_fetch_bytes=config.kafka_max_message_size,
+        )
+
+        try:
+            await consumer.start()
```

Comment:
```suggestion
        await consumer.start()
        try:
```

Reply:
If the consumer was not started, it should not be closed.

Reply:
Fixed in [4d2b5c9](https://github.com/ghga-de/hexkit/pull/123/commits/4d2b5c920f4a048d97cb147a5a3e772af03abee8)